client/client.go

```diff
@@ -127,12 +141,102 @@ func (client Mikrotik) UpdateDnsRecord(id, name, address string, ttl int) error
 
 func (client Mikrotik) DeleteDnsRecord(id string) error {
 	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return err
+	}
 	cmd := strings.Split(fmt.Sprintf("/ip/dns/static/remove =numbers=%s", id), " ")
 	fmt.Println(fmt.Sprintf("[INFO] Running the mikrotik command: `%s`", cmd))
 	_, err = c.RunArgs(cmd)
 	return err
 }
 
+func (client Mikrotik) AddDhcpLease(address, macaddress string) (*routeros.Reply, error) {
+	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return nil, err
+	}
+	cmd := strings.Split(fmt.Sprintf("/ip/dhcp-server/lease/add =address=%s =mac-address=%s", address, macaddress), " ")
+	fmt.Println(fmt.Sprintf("[INFO] Running the mikrotik command: `%s`", cmd))
+	r, err := c.RunArgs(cmd)
+	return r, err
+}
+
+func (client Mikrotik) FindDhcpLease(address string) (*DhcpLease, error) {
+	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return nil, err
+	}
+	cmd := "/ip/dhcp-server/lease/print"
+	fmt.Println(fmt.Sprintf("[INFO] Running the mikrotik command: `%s`", cmd))
```

Comment:
As shown in #9 this should be `log.Printf`

Reply:
Updated